compiler/luci/pass/src/FuseRmsNormPass.cpp

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseRmsNormPass.h"
+#include "helpers/NodeFiller.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/CircleNodeClone.h>
+
+#include <cmath>
+#include <cassert>
+
+namespace
+{
+
+/**
+ * Below diagram shows RMS normalization pattern to fuse.
+ * - this pattern will be replaced with one RmsNorm
+ *
+ *           [In]
+ *            |
+ *            V
+ *     +---- ifm ----+
+ *     |      |      |
+ *     |      V      |
+ *     |     mul <---+
+ *     |      |
+ *     |      V
+ *     |     mean
+ *     |      |
+ *     |      V
+ *     |     add_epsilon
+ *     |      |
+ *     |      V
+ *     |     rsqrt
+ *     |      |
+ *     |      V
+ *     +---> mul_input
+ *            |
+ *            V
+ *          [Out]
+ */
+
+class RmsNormPattern final
+{
+public:
+  RmsNormPattern(luci::CircleMul *candidate)
+  {
+    assert(candidate); // FIX_CALLER_UNLESS
+    _mul_input = candidate;
+  }
+
+public:
+  bool matched();
+
+public:
+  luci::CircleNode *_ifm = nullptr;
+  luci::CircleMul *_mul_pow = nullptr;
+  luci::CircleMean *_mean = nullptr;
+  luci::CircleAdd *_add_epsilon = nullptr;
+  luci::CircleRsqrt *_rsqrt = nullptr;
+  luci::CircleMul *_mul_input = nullptr;
+  luci::CircleMul *_mul_gamma = nullptr; // not used
+  luci::CircleAdd *_add_beta = nullptr;  // not used
+  luci::CircleConst *_const_epsilon = nullptr;
+  luci::CircleConst *_const_gamma = nullptr; // use default gamma(scale) 1.0
+  luci::CircleConst *_const_beta = nullptr;  // use default beta(offset) 0.0
+};
+
+#define CHECK_OR_FALSE(condition) \
+  if (not(condition))             \
+    return false;
+
+luci::CircleConst *make_const_one(loco::Graph *graph, float value)
+{
+  auto const_one = graph->nodes()->create<luci::CircleConst>();
+  const_one->dtype(loco::DataType::FLOAT32);
+  const_one->rank(1);
+  const_one->size<loco::DataType::FLOAT32>(1);
+  const_one->at<loco::DataType::FLOAT32>(0) = value;
+  return const_one;
+}
+
+bool RmsNormPattern::matched()
+{
+  // Check pattern
```

Comment:
IMHO, this comment can be removed as it doesn't give particular help reading the code.

Reply:
Okay, I'll update it.